src/key.js

```diff
@@ -1380,6 +1336,43 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPacket, options) {
   return new Key(packetlist);
 }
 
+/**
+ * Checks if a given certificate or binding signature is revoked
+ * @param  {module:packet/secret_key|
+ *          module:packet/public_key}       primaryKey   The primary key packet
+ * @param  {Object}                         dataToVerify The data to check
+ * @param  {Array<module:packet/signature>} revocations  The revocation signatures to check
+ * @param  {module:packet/signature}        signature    The certificate or signature to check
+ * @param  {module:packet/public_subkey|
+ *          module:packet/secret_subkey|
+ *          module:packet/public_key|
+ *          module:packet/secret_key} key, optional The key packet to check the signature
+ * @param  {Date}                     date          Use the given date instead of the current time
+ * @return {Boolean}                                True if the signature revokes the data
+ */
+async function isDataRevoked(primaryKey, dataToVerify, revocations, signature, key, date=new Date()) {
+  key = key || primaryKey;
+  const normDate = util.normalizeDate(date);
+  const revocationKeyIds = [];
+  await Promise.all(revocations.map(async function(revocationSignature) {
+    if (!(config.revocations_expire && revocationSignature.isExpired(normDate)) &&
+        (revocationSignature.verified || await revocationSignature.verify(key, dataToVerify))) {
+      // TODO get an identifier of the revoked object instead
```

Comment:
This is a bug in the previous implementation: if you pass a random certificate and ask whether this certificate has been revoked by a revocation signature, after validating the signature, the code simply checks whether the certificate and the signature have the same issuerKeyId. So really the certificate could have been a completely irrelevant certificate!

For now I just put a TODO because a user who is working with high level API's wouldn't run into this problem. The correct solution is something like checking that the hash of the data which was signed is the same, but I couldn't find the correct variables.